src/libs/relay/conduit_relay_mpi.cpp

```diff
@@ -113,171 +113,618 @@ rank(MPI_Comm mpi_comm)
+
+        #if defined(CONDUIT_USE_LONG_LONG)
+        case CONDUIT_NATIVE_LONG_LONG_ID: res = MPI_LONG_LONG; break;
+        #endif
+
+        // unsigned integer types 
+        case CONDUIT_NATIVE_UNSIGNED_CHAR_ID:  res = MPI_UNSIGNED_CHAR;  break;
+        case CONDUIT_NATIVE_UNSIGNED_SHORT_ID: res = MPI_UNSIGNED_SHORT; break;
+        case CONDUIT_NATIVE_UNSIGNED_INT_ID:   res = MPI_UNSIGNED;       break;
+        case CONDUIT_NATIVE_UNSIGNED_LONG_ID:  res = MPI_UNSIGNED_LONG;  break;
+        
+        #if defined(CONDUIT_USE_LONG_LONG)
+        case CONDUIT_NATIVE_UNSIGNED_LONG_LONG_ID:
+        {
+            res = MPI_UNSIGNED_LONG_LONG;
+            break;
+        }
+        #endif
+
+        // floating point types
+
+        case CONDUIT_NATIVE_FLOAT_ID:  res = MPI_FLOAT;  break;
+        case CONDUIT_NATIVE_DOUBLE_ID: res = MPI_DOUBLE; break;
+
+        #if defined(CONDUIT_USE_LONG_DOUBLE)
+        case CONDUIT_NATIVE_LONG_DOUBLE_ID: res = MPI_LONG_DOUBLE; break;
+        #endif
+
+
+        case DataType::CHAR8_STR_ID:  res = MPI_CHAR;  break;
+
+    }
+    
+    return res;
+}
+
+
+//-----------------------------------------------------------------------------
+index_t
+mpi_dtype_to_conduit_dtype_id(MPI_Datatype dt)
+{
+    index_t res = DataType::EMPTY_ID;
+
+    switch(dt)
+    {
+        // string type
+        case MPI_CHAR:  res = DataType::CHAR8_STR_ID;  break;
+
+        // signed integer types
+        case MPI_SHORT: res = CONDUIT_NATIVE_SHORT_ID; break;
+        case MPI_INT:   res = CONDUIT_NATIVE_INT_ID;   break;
+        case MPI_LONG:  res = CONDUIT_NATIVE_LONG_ID;  break;
+
+        #if defined(CONDUIT_USE_LONG_LONG)
+        case MPI_LONG_LONG: res = CONDUIT_NATIVE_LONG_LONG_ID; break;
+        #endif
+
+        // unsigned integer types 
+        case MPI_BYTE:           res = CONDUIT_NATIVE_UNSIGNED_CHAR_ID;  break;
+        case MPI_UNSIGNED_CHAR:  res = CONDUIT_NATIVE_UNSIGNED_CHAR_ID;  break;
+        case MPI_UNSIGNED_SHORT: res = CONDUIT_NATIVE_UNSIGNED_SHORT_ID; break;
+        case MPI_UNSIGNED:       res = CONDUIT_NATIVE_UNSIGNED_INT_ID;   break;
+        case MPI_UNSIGNED_LONG:  res = CONDUIT_NATIVE_UNSIGNED_LONG_ID;  break;
+        
+        #if defined(CONDUIT_USE_LONG_LONG)
+        case MPI_UNSIGNED_LONG_LONG:
+        {
+            res = CONDUIT_NATIVE_UNSIGNED_LONG_LONG_ID;
+            break;
+        }
+        #endif
+
+        // floating point types
+
+        case MPI_FLOAT:  res = CONDUIT_NATIVE_FLOAT_ID;  break;
+        case MPI_DOUBLE: res = CONDUIT_NATIVE_DOUBLE_ID; break;
+
+        #if defined(CONDUIT_USE_LONG_DOUBLE)
+        case MPI_LONG_DOUBLE: res = CONDUIT_NATIVE_LONG_DOUBLE_ID; break;
+        #endif
+
+    }
+
+    return res;
+}
+
 
 //---------------------------------------------------------------------------//
 int 
-send(Node &node, int dest, int tag, MPI_Comm comm)
+send_using_schema(const Node &node, int dest, int tag, MPI_Comm comm)
```

Comment:
I'll vote for everything to be using_schemas. Technically, you have a schema on both sides, even if they're the same one.